File: ObjectBrain.Cart/Presentation.Web/wwwroot/app/Service/ICartClientService.ts
import { Injectable }    from '@angular/core';
import {EntityBase} from '../Model/EntityBase';

@Injectable()
export interface ICartClientService<T extends EntityBase> {

    GetAll(): Promise<Array<T>>;
    Get(id: number): Promise<T>;
    Add(entity: T);
    Update(entity: T);
    Remove(entity: T);
}

File: ObjectBrain.Cart/Presentation.Wpf/Service/ICartClientService.cs
using System;
using System.Collections.Generic;

namespace Presentation.Wpf.Service
{
    public interface ICartClientService<T>
    {
        IObservable<IEnumerable<T>> GetAll();
        IObservable<T> Get(int id);
        void Add(T entity);
        void Update(T entity);
        void Remove(T entity);
    }
}